app/assets/javascripts/views/shared/quickLinksView.js

```diff
@@ -37,6 +37,21 @@
     initialize: function (settings) {
       this.options = Object.assign({}, this.defaults, settings);
 
+      let siteSlug = window.location.href.match(/\/sites\/(\S+)\//);
+      siteSlug = siteSlug ? siteSlug[1] : '';
+      let userSiteRole = gon.global.user.user_site_roles.filter((userSite) => {
+        return userSite.site_slug === siteSlug;
+      });
+      userSiteRole = userSiteRole.length ? userSiteRole[0].role : "";
+      if (userSiteRole === 'publisher') {
+        for(var i = this.options.links.length; i--;) {
+          if (this.options.links[i]['id'] === 'admin' ||
+              this.options.links[i]['id'] === 'contexts') {
+            this.options.links.splice(i, 1);
+          }
```

Comment:
You could simplify the code by this:
```js
this.options.links = this.options.links.filter(link => link.id !== 'admin' && link.id !== 'contexts');
```

But feel free to keep as it is 😄.

Reply:
Too much time without using javascript 💯